beacon-chain/sync/rpc_status.go

```diff
@@ -275,30 +300,123 @@ func (s *Service) statusRPCHandler(ctx context.Context, msg interface{}, stream
 func (s *Service) respondWithStatus(ctx context.Context, stream network.Stream) error {
 	headRoot, err := s.cfg.chain.HeadRoot(ctx)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "chain head root")
 	}
 
 	forkDigest, err := s.currentForkDigest()
 	if err != nil {
-		return err
+		return errors.Wrap(err, "current fork digest")
 	}
+
 	cp := s.cfg.chain.FinalizedCheckpt()
-	resp := &pb.Status{
+	status, err := s.buildStatusFromStream(stream, forkDigest, cp.Root, cp.Epoch, headRoot)
+	if err != nil {
+		return errors.Wrap(err, "build status")
+	}
+
+	if _, err := stream.Write([]byte{responseCodeSuccess}); err != nil && !isUnwantedError(err) {
+		log.WithError(err).Debug("Could not write to stream")
+	}
+
+	if _, err := s.cfg.p2p.Encoding().EncodeWithMaxLength(stream, status); err != nil {
+		return errors.Wrap(err, "encode with max length")
+	}
+
+	return nil
+}
+
+func (s *Service) buildStatusFromStream(
+	stream libp2pcore.Stream,
+	forkDigest [4]byte,
+	finalizedRoot []byte,
+	FinalizedEpoch primitives.Epoch,
+	headRoot []byte,
+) (ssz.Marshaler, error) {
+	// Get the stream version from the protocol.
+	_, _, streamVersion, err := p2p.TopicDeconstructor(string(stream.Protocol()))
+	if err != nil {
+		err := errors.Wrap(err, "topic deconstructor")
+
+		resp, err2 := s.generateErrorResponse(responseCodeServerError, types.ErrGeneric.Error())
+		if err2 != nil {
+			log.WithError(err2).Debug("Could not write to stream")
+			return nil, err
+		}
+
+		if _, err2 := stream.Write(resp); err != nil {
+			log.WithError(err2).Debug("Could not write to stream")
+		}
+
+		return nil, err
+	}
+
+	earliestAvailableSlot, err := s.cfg.p2p.EarliestAvailableSlot()
+	if err != nil {
+		return nil, errors.Wrap(err, "earliest available slot")
+	}
+
+	if streamVersion == p2p.SchemaVersionV2 {
```

Comment:
should we have a check for streamVersion isn't any of these v2 or v1? have a switch statement to handle?

Reply:
don't we also have SchemaVersionV3?

Reply:
> should we have a check for streamVersion isn't any of these v2 or v1? have a switch statement to handle?

No need actually, such a check is already done here:
https://github.com/OffchainLabs/prysm/blob/2e4ba050f756aa603b531f486dd389b50fe5d7a9/beacon-chain/sync/rpc_status.go#L490-L509

Reply:
> don't we also have SchemaVersionV3?

Not I am aware of. Where did you see that?

Reply:
> don't we also have SchemaVersionV3?

> Not I am aware of. Where did you see that?

https://github.com/OffchainLabs/prysm/blob/e8625cd89d730f80222ed6b64b44f9b515e4bf7f/beacon-chain/p2p/rpc_topic_mappings.go#L22

this, i think there's some confusion as schemaVersion and status V2 is different right?